File: Frontend/ris-frontend/src/components/Doctor.js
import React, { useState, useEffect } from 'react';
import axios from 'axios';

const Doctor = () => {
  const [doctors, setDoctors] = useState([]); // Inicializamos como array vacío
  const [name, setName] = useState('');
  const [lastName, setLastName] = useState('');
  const [curp, setCurp] = useState('');
  const [phone, setPhone] = useState('');
  const [age, setAge] = useState('');
  const [speciality, setSpeciality] = useState('');
  const [status, setStatus] = useState(true);

  useEffect(() => {
    fetchDoctors();
  }, []);

  const fetchDoctors = async () => {
    try {
      const response = await axios.get('http://localhost:8080/doctor/all');
      setDoctors(response.data.data || []); // Aseguramos que no sea undefined
    } catch (error) {
      console.error('Error fetching doctors:', error);
    }
  };

  const handleAddDoctor = async () => {
    const doctorData = { name, lastName, curp, phone, age: parseInt(age), speciality, status };
    try {
      const response = await axios.post('http://localhost:8080/doctor/add', doctorData);
      console.log('Doctor added:', response.data);
      fetchDoctors();
      clearForm();
    } catch (error) {
      console.error('Error adding doctor:', error);
    }
  };

  const handleDeleteDoctor = async (id) => {
    try {
      await axios.delete(`http://localhost:8080/doctor/delete/${id}`);
      fetchDoctors();
    } catch (error) {
      console.error('Error deleting doctor:', error);
    }
  };

  const handleEditDoctor = (doctor) => {
    setName(doctor.name);
    setLastName(doctor.lastName);
    setCurp(doctor.curp);
    setPhone(doctor.phone);
    setAge(doctor.age);
    setSpeciality(doctor.speciality);
    setStatus(doctor.status);
  };

  const clearForm = () => {
    setName('');
    setLastName('');
    setCurp('');
    setPhone('');
    setAge('');
    setSpeciality('');
    setStatus(true);
  };

  return (
    <div>
      <h2>Doctors</h2>

      <form onSubmit={(e) => e.preventDefault()}>
        <input type="text" placeholder="Name" value={name} onChange={(e) => setName(e.target.value)} />
        <input type="text" placeholder="Last Name" value={lastName} onChange={(e) => setLastName(e.target.value)} />
        <input type="text" placeholder="CURP" value={curp} onChange={(e) => setCurp(e.target.value)} />
        <input type="text" placeholder="Phone" value={phone} onChange={(e) => setPhone(e.target.value)} />
        <input type="number" placeholder="Age" value={age} onChange={(e) => setAge(e.target.value)} />
        <input type="text" placeholder="Specialty" value={speciality} onChange={(e) => setSpeciality(e.target.value)} />
        <label>
          Active:
          <input type="checkbox" checked={status} onChange={() => setStatus(!status)} />
        </label>
        <button type="button" onClick={handleAddDoctor}>Add Doctor</button>
      </form>

      <ul>
        {doctors && doctors.length > 0 ? (
          doctors.map((doctor) => (
            <li key={doctor.id}>
              {doctor.name} {doctor.lastName} - {doctor.speciality}
              <button onClick={() => handleEditDoctor(doctor)}>Edit</button>
              <button onClick={() => handleDeleteDoctor(doctor.id)}>Delete</button>
            </li>
          ))
        ) : (
          <li>No doctors available</li>
        )}
      </ul>
    </div>
  );
};

export default Doctor;

File: src/main/java/com/example/demo/doctor/model/Doctor.java
package com.example.demo.Doctor.model;

import jakarta.persistence.*;

@Entity
@Table(name = "doctors")
public class Doctor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name",columnDefinition = "VARCHAR(30)")
    private String name;

    @Column(name = "last_name",columnDefinition = "VARCHAR(30)")
    private String lastName;

    @Column(name = "curp",columnDefinition = "VARCHAR(18)")
    private String curp;

    @Column(name = "phone",columnDefinition = "VARCHAR(15)")
    private String phone;


    @Column(name = "age",columnDefinition = "INTEGER")
    private int age;
    @Column(name = "speciality",columnDefinition = "VARCHAR(30)")
    private String speciality;

    @Column(name = "status",columnDefinition = "BOOL DEFAULT TRUE")
    private boolean status;

    public Doctor() {
    }

    public Doctor(String name, String speciality) {
        this.name = name;
        this.speciality = speciality;
    }

    public Doctor(Long id, String name, String lastName, String curp, String phone, int age, String speciality, boolean status) {
        this.id = id;
        this.name = name;
        this.lastName = lastName;
        this.curp = curp;
        this.phone = phone;
        this.age = age;
        this.speciality = speciality;
        this.status = status;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getSpeciality() {
        return speciality;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setSpeciality(String speciality) {
        this.speciality = speciality;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getCurp() {
        return curp;
    }

    public void setCurp(String curp) {
        this.curp = curp;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }
}
